src/summary/patient-chart-summary.component.tsx

```diff
@@ -10,44 +10,92 @@ import ProgramsCard from "./history/programs/programs-card.component";
 import MedicationsSummary from "./history/medications/medications-summary.component";
 
 export default function PatientChartSummary(props: PatientChartSummaryProps) {
-  const config = [
-    "conditions",
-    "programs",
-    "medications",
-    "allergies",
-    "notes",
-    "vitals",
-    "heightAndWeight"
+  const exampleConfig: configType[] = [
+    { name: "conditions" },
+    {
+      name: "Programs",
+      esModule: "@jj-widgets",
+      exportName: "programsWidget"
+    },
+    { name: "medications" },
+    { name: "allergies" },
+    { name: "notes" },
+    { name: "vitals" },
+    { name: "heightAndWeight" }
   ];
 
-  const coreComponents = {
-    conditions: ConditionsCard,
-    programs: ProgramsCard,
-    allergies: AllergyCard,
-    notes: NotesCard,
-    vitals: VitalsCard,
-    heightAndWeight: DimensionsCard,
-    medications: MedicationsSummary
-  };
+  const config: configType[] = [
+    { name: "conditions" },
+    { name: "medications" },
+    { name: "allergies" },
+    { name: "notes" },
+    { name: "vitals" },
+    { name: "heightAndWeight" },
+    { name: "programs" }
+  ];
+
+  const [widgets, setWidgets] = React.useState([]);
+
+  React.useEffect(() => {
+    const coreComponents = {
+      conditions: ConditionsCard,
+      programs: ProgramsCard,
+      allergies: AllergyCard,
+      notes: NotesCard,
+      vitals: VitalsCard,
+      heightAndWeight: DimensionsCard,
+      medications: MedicationsSummary
+    };
+
+    const modulePromises = [];
```

Comment:
```suggestion
    // TODO: Refactor this so that these promises only happen once. 
    //   Probably just in the root component, and then the router passes the widgets to the pages.
    const modulePromises = [];
```

Reply:
My understanding is that system.js will cache after importing. In subsequent, calls, I think of it (and this may be wrong) as more akin to doing a standard "import" although you're right, it does return a promise. From a performance perspective, I think it's safe. I'm also not sure how we would "save" the imports after being load in the root component. We could make them a prop to a child of root but that seems messy to me. It seemed cleaner to just use system.import again. What do you think?

Reply:
If you're right about caching, then your solution seems perfect :+1: